systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java

```diff
@@ -366,6 +366,34 @@ void testKafkaMirrorMaker2WrongBootstrap() {
         KafkaMirrorMaker2Resource.deleteKafkaMirrorMaker2WithoutWait(kafkaMirrorMaker2);
     }
 
+    @Test
+    void testKafkaTopicDecreaseStatus() {
+        KafkaTopicResource.topic(CLUSTER_NAME, TEST_TOPIC_NAME, 5).done();
+        int decreaseTo = 1;
+
+        LOGGER.info("Decreasing number of partitions to {}", decreaseTo);
+        KafkaTopicResource.replaceTopicResource(TEST_TOPIC_NAME, kafkaTopic -> kafkaTopic.getSpec().setPartitions(decreaseTo));
+        KafkaTopicUtils.waitForKafkaTopicPartitionChange(TEST_TOPIC_NAME, decreaseTo);
+
+        assertKafkaTopicDecreasePartitionsStatus(TEST_TOPIC_NAME);
+    }
+
+    @Test
+    void testKafkaTopicChangingInSyncReplicasStatus() throws InterruptedException {
+        KafkaTopicResource.topic(CLUSTER_NAME, TEST_TOPIC_NAME, 5).done();
+        String invalidValue = "x";
+
+        LOGGER.info("Changing min.insync.replicas to random char");
+        KafkaTopicResource.replaceTopicResource(TEST_TOPIC_NAME, kafkaTopic -> kafkaTopic.getSpec().getConfig().replace("min.insync.replicas", invalidValue));
+        KafkaTopicUtils.waitForKafkaTopicNotReady(TEST_TOPIC_NAME);
+
+        assertKafkaTopicWrongMinInSyncReplicasStatus(TEST_TOPIC_NAME, invalidValue);
+
+        // Wait some time to check if error is still present in KafkaTopic status
+        Thread.sleep(60_000);
```

Comment:
It's the reconciliation interval you need to wait for. Hard coding 60s is fragile. You could obtaint the interval by reading it from the env var of the TO container in the EO pod. That way the test would not break even if the interval later got changed.

Reply:
Thanks for comment, I'll fix it